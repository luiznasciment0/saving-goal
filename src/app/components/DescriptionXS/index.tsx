import Typography from 'app/components/Typography'

interface DescriptionXSProps {
  children: React.ReactNode
}

const DescriptionXS = ({ children }: DescriptionXSProps) => (
  <Typography as="h4" size="xsmall" weight="light" color="primaryBlack">
    {children}
  </Typography>
)

export default DescriptionXS
