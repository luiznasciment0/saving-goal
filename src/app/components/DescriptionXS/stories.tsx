import { Story, Meta } from '@storybook/react/types-6-0'
import DescriptionXS from '.'

export default {
  title: 'DescriptionXS',
  component: DescriptionXS,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story = (args) => (
  <DescriptionXS>{args.children}</DescriptionXS>
)

Default.args = {
  children: 'Teste'
}
