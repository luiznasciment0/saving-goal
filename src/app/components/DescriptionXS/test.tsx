import { render, screen } from '@testing-library/react'

import DescriptionXS from '.'

describe('<DescriptionXS />', () => {
  it('should render the DescriptionXS and its styles', () => {
    const { container } = render(<DescriptionXS>Teste</DescriptionXS>)

    expect(screen.getByRole('heading', { name: /teste/i })).toHaveStyle({
      'font-size': '1.2rem',
      'line-height': '1.6rem',
      'font-weight': '400',
      'font-family': `'Work Sans'`,
      color: '#1e2a32'
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
