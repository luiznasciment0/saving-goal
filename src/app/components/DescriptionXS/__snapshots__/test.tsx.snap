// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<DescriptionXS /> should render the DescriptionXS and its styles 1`
] = `
.c0 {
  font-size: 1.2rem;
  line-height: 1.6rem;
  font-family: 'Work Sans';
  color: #1e2a32;
  font-weight: 400;
}

<h4
  class="c0"
  color="primaryBlack"
>
  Teste
</h4>
`
